content/browser/adblock_interceptor.cc

```diff
@@ -0,0 +1,95 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/content/browser/adblock_interceptor.h"
+
+#include "base/bind.h"
+#include "base/compiler_specific.h"
+#include "base/location.h"
+#include "base/single_thread_task_runner.h"
+#include "base/threading/thread_task_runner_handle.h"
+#include "base/memory/weak_ptr.h"
+#include "net/url_request/url_request_job.h"
+#include "net/http/http_response_info.h"
+#include "net/http/http_response_headers.h"
+#include "net/http/http_util.h"
+
+namespace brave {
+namespace content {
+
+class Http200OkJob : public net::URLRequestJob {
+ public:
+  Http200OkJob(net::URLRequest* request,
+               net::NetworkDelegate* network_delegate);
+
+  // URLRequestJob:
+  void Start() override;
+  void Kill() override;
+  bool GetMimeType(std::string* mime_type) const override;
+  void GetResponseInfo(net::HttpResponseInfo* info) override;
+
+ private:
+  ~Http200OkJob() override;
+  void StartAsync();
+
+  base::WeakPtrFactory<Http200OkJob> weak_factory_;
+};
+
+Http200OkJob::Http200OkJob(net::URLRequest* request,
+                           net::NetworkDelegate* network_delegate)
+    : net::URLRequestJob(request, network_delegate), weak_factory_(this) {}
+
+void Http200OkJob::Start()  {
+  // Start reading asynchronously so that all error reporting and data
+  // callbacks happen as they would for network requests.
+  base::ThreadTaskRunnerHandle::Get()->PostTask(
+      FROM_HERE, base::BindOnce(&Http200OkJob::StartAsync,
+                                weak_factory_.GetWeakPtr()));
+}
+
+void Http200OkJob::Kill() {
+  weak_factory_.InvalidateWeakPtrs();
+  URLRequestJob::Kill();
+}
+
+bool Http200OkJob::GetMimeType(std::string* mime_type) const {
+  *mime_type = "text/html";
```

Comment:
this seems potentially problematic, especially since most of these requests will be for JS. Shouldn't we be looking at the request headers?

Reply:
Well, it seems less problematic than current solution... I quickly tested it and it seems OK.
How exactly headers could be analyzed?

Reply:
the accept header?

Reply:
we could also use the file type in the request. I mean, maybe it's fine, but seems like it would be better to try to match the content type

Reply:
or at the very least return something that is in the accept header

Reply:
From what I could see, typically `Accept` is not present on ad/tracker requests :-/ Typically it is used for image requests. I can make this improvement, though don't think it is critical.

Reply:
done